### PR TITLE
Application acceptance and rejection

### LINKED ISSUE #21: User Story 12

### LINKED ISSUE #22: User Story 13

### PR DESCRIPTION
**Pull request by noahdurbin**
This pull request adds functionality to approve or reject pets for adoption within the admin application show page. 
- Added routes for the admin application show
- added migration for status, including giving status a default value of pending
- created an admin applications controller
- added logic to update the pet applications status
- updated view to show the status rather than the buttons for acceptance or rejection
- Added feature tests to cover the approval and rejection

### CODE REVIEW

File: app/controllers/applications_controller.rb [hunk 1]
```diff
@@ -4,14 +4,12 @@ def show
     @pets = Pet.search(params[:pet_name]) if params[:pet_name]
   end
 
-  def new
-    # @application = Application.new
```
Comment (AuroraTuring) [inline app/controllers/applications_controller.rb:8]: Good housekeeping.

File: db/migrate/20240522034351_add_status_to_pet_applications.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToPetApplications < ActiveRecord::Migration[7.1]
+  def change
+    add_column :pet_applications, :status, :string, default: :Pending, null: false
```
Comment (AuroraTuring) [inline db/migrate/20240522034351_add_status_to_pet_applications.rb:3]: Adding the status column was helpful for us to be able to retrieve it as an attribute later on!

File: spec/features/admin/shelters/index_spec.rb [hunk 3]
```diff
@@ -2,32 +2,28 @@
...
-    @shelter1 = Shelter.create!(foster_program: true, name: "Whimsy Whiskers", city: "Denver", rank: 42)
-    @shelter2 = Shelter.create!(foster_program: true, name: "Meow Wows", city: "New York", rank: 42)
-    @shelter3 = Shelter.create!(foster_program: false, name: "Sandy Meows", city: "Miami", rank: 42)
+    @shelter1 = Shelter.create!(foster_program: true, name: 'Whimsy Whiskers', city: 'Denver', rank: 42)
+    @shelter2 = Shelter.create!(foster_program: true, name: 'Meow Wows', city: 'New York', rank: 42)
+    @shelter3 = Shelter.create!(foster_program: false, name: 'Sandy Meows', city: 'Miami', rank: 42)
   end
 
-  # As a visitor
-  # When I visit the admin shelter index ('/admin/shelters')
```
Comment (AuroraTuring) [inline spec/features/admin/shelters/index_spec.rb:11]: Good housekeeping again.

File: app/views/admin/applications/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  <%= pet.name %>
+<% end %>
+<br>Application Status: <%= @application.status %></p>
+
+<% if @application.pets.count > 0 %>
+  <br><h2>All Pets: </h2>
+  <% @application.pet_applications.each do |pet_app| %>
+    <%= pet_app.pet.name %>
+    <% if pet_app.status == "Pending" %>
+      <%= button_to "Reject", "/admin/applications/#{@application.id}?status=Rejected", method: :patch, params: {pet_application_id: pet_app.id} %>
```
Comment (JCL461437) [inline app/views/admin/applications/show.html.erb:14]: Wow! These buttons are great!

File: app/models/pet_application.rb [hunk 5]
```diff
@@ -1,4 +1,8 @@
 class PetApplication < ApplicationRecord
   belongs_to :application
   belongs_to :pet
+
+  def self.by_application_id(application_id)
```
Comment (JCL461437) [inline app/models/pet_application.rb:5]: This is an interesting method. I think this is an efficient way to be able to query whether or not the id matches the application id we are looking for

File: spec/features/applications/show_spec.rb [hunk 6]
```diff
@@ -117,7 +108,7 @@
           fill_in 'pet_name', with: 'bUd'
```
Comment (AuroraTuring) [inline spec/features/applications/show_spec.rb:108]: Good second test to ensure case insensitivity.

File: config/routes.rb [hunk 7]
```diff
@@ -2,6 +2,8 @@
   get '/', to: 'application#welcome'
 
   get '/admin/shelters', to: 'admin/shelters#index'
```
Comment (AuroraTuring) [inline config/routes.rb:4]: Good job creating admin specific routes!

File: spec/features/admin/applications/show_spec.rb [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    @application1 = Application.create!(name: 'John Doe', address: '123 Elm St', city: 'Denver', state: 'CO',
+                                        zip: '12345', description: 'Looking for a friendly dog', status: 'In Progress')
+    PetApplication.create!(pet: @pet1, application: @application1, status: :Pending)
+  end
+
+  describe 'as a visitor' do
+    it 'there is a button to reject an application' do
+      visit "/admin/applications/#{@application1.id}"
+
+      expect(page).to have_content("Reject")
```
Comment (AuroraTuring) [inline spec/features/admin/applications/show_spec.rb:16]: Nice simple test to check for content.

File: spec/features/admin/applications/show_spec.rb [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+      visit "/admin/applications/#{@application1.id}"
+
+      expect(page).to have_content("Approve")
+    end
+
+    it 'when a pet is Approved on an application there is a status of Approved and button to Approve is gone' do
+      visit "/admin/applications/#{@application1.id}"
+      
+      expect(page).to_not have_content('Approved')
+      click_button 'Approve'
```
Comment (AuroraTuring) [inline spec/features/admin/applications/show_spec.rb:38]: Nice execution.

File: app/views/admin/applications/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+      <%= button_to "Approve", "/admin/applications/#{@application.id}?status=Approved", method: :patch, params: {pet_application_id: pet_app.id} %>
+    <% else %>
+      <p><%= pet_app.status.to_s %></p>
```
Comment (JCL461437) [inline app/views/admin/applications/show.html.erb:17]: I like the presentation logic utilized with the integrated ruby conditional

**Review by AuroraTuring — APPROVED** [review]

File: spec/features/admin/applications/show_spec.rb [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+      expect(page).to have_content('Rejected')
+    end
+
+    it 'there is a button to approve an application' do
+      visit "/admin/applications/#{@application1.id}"
+
+      expect(page).to have_content("Approve")
+    end
+
+    it 'when a pet is Approved on an application there is a status of Approved and button to Approve is gone' do
```
Comment (JCL461437) [inline spec/features/admin/applications/show_spec.rb:34]: I think these are good robust tests. Great work Noah

**Review by JCL461437 — APPROVED** [review]

**Pull request merged**